### ISSUE TITLE
Bug: Shell becomes unresponsive after quitting local web framework hosting programs

### ISSUE THREAD

**Issue by 2gn** [holzschu/a-shell#700]
## Steps to reproduce
1. create an index.html file
2. run something like this on the same directory as the html file

```python3
#! /usr/bin/env python3

from subprocess import run as sp_run
from bottle import route, static_file, run

@route('/')
def root():
    return static_file('index.html', root='.')

if __name__ == '__main__':
    run(host='localhost', port=8080)
```

3. (optional) access http://localhost:8080 (this works fine)

4. try to quit the program by pressing Ctrl-C and Ctrl-D for several times

5. The program quits, but the shell becomes unresponsive 

Thanks.

**Comment by holzschu** [comment]
Basically, anything involving subprocess is going to cause problems like this. There is no easy way to decide who should get the keyboard input, how to stop the subprocess, all these things.

**Comment by Iamrodos** [comment]
I am running a Flask app and have the same issue. Can't terminate the python program and have to kill/restart the a-shell.

**Comment by holzschu** [comment]
This sort of things (running a web server inside a-Shell) is always going to be tricky. a-Shell does not have the possibility to kill a running command (or, if it does, that leaves the app in a bad state), it has to tell it politely to terminate itself and to clean up everything. That only works if the command is actively listening. A Flask app is actively listening to a lot of things, but mostly to the commands coming from the web page it is serving. It also does not have a routine to clean up everything (close the sockets, free the memory) unless you explicitly added one.

Short version: killing a-Shell is probably the safest way to terminate a Flask app. If that is an issue, the second best thing would be to add a "Quit" button on the web page, and connect it to a routine that closes the sockets, stops the server and terminates Python.
